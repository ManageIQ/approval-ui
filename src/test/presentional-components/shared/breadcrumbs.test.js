import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import ApprovalBreadcrumbs from '../../../presentational-components/shared/breadcrubms';

describe('<ApprovalBreadcrumbs />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      breadcrumbs: [{
        title: 'First level',
        to: '/first-level'
      }, {
        title: 'Second level',
        isActive: true
      }]
    };
  });

  it('should render breadcrumbs', () => {
    const wrapper = mount(
      <MemoryRouter>
        <ApprovalBreadcrumbs { ...initialProps } />
      </MemoryRouter>
    );

    expect(toJson(wrapper.find(ApprovalBreadcrumbs))).toMatchSnapshot();
  });

  it('should not render if no data is provided', () => {
    const wrapper = mount(
      <MemoryRouter>
        <ApprovalBreadcrumbs />
      </MemoryRouter>
    );

    expect(toJson(wrapper.find(ApprovalBreadcrumbs))).toMatchSnapshot();
  });
});
