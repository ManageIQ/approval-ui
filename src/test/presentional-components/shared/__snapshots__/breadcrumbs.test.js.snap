// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApprovalBreadcrumbs /> should not render if no data is provided 1`] = `<ApprovalBreadcrumbs />`;

exports[`<ApprovalBreadcrumbs /> should render breadcrumbs 1`] = `
<ApprovalBreadcrumbs
  breadcrumbs={
    Array [
      Object {
        "title": "First level",
        "to": "/first-level",
      },
      Object {
        "isActive": true,
        "title": "Second level",
      },
    ]
  }
>
  <Component>
    <ComponentWithOuia
      component={[Function]}
      componentProps={
        Object {
          "children": Array [
            <BreadcrumbItem
              isActive={false}
            >
              <ForwardRef
                exact={true}
                isActive={[Function]}
                to="/first-level"
              >
                First level
              </ForwardRef>
            </BreadcrumbItem>,
            <BreadcrumbItem
              isActive={true}
            >
              Second level
            </BreadcrumbItem>,
          ],
        }
      }
      consumerContext={null}
    >
      <Breadcrumb
        ouiaContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
      >
        <nav
          aria-label="Breadcrumb"
          className="pf-c-breadcrumb"
        >
          <ol
            className="pf-c-breadcrumb__list"
          >
            <BreadcrumbItem
              isActive={false}
              key="First level"
            >
              <li
                className="pf-c-breadcrumb__item"
              >
                <NavLink
                  exact={true}
                  isActive={[Function]}
                  to="/first-level"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/first-level",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/first-level"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/first-level"
                        onClick={[Function]}
                      >
                        First level
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
                <span
                  className="pf-c-breadcrumb__item-divider"
                >
                  <AngleRightIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                    title={null}
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        transform=""
                      />
                    </svg>
                  </AngleRightIcon>
                </span>
              </li>
            </BreadcrumbItem>
            <BreadcrumbItem
              isActive={true}
              key="Second level"
            >
              <li
                className="pf-c-breadcrumb__item"
              >
                Second level
              </li>
            </BreadcrumbItem>
          </ol>
        </nav>
      </Breadcrumb>
    </ComponentWithOuia>
  </Component>
</ApprovalBreadcrumbs>
`;
